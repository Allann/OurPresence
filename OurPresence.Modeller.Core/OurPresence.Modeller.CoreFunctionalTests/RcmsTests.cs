// Copyright (c)  Allan Nielsen.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
using ApprovalTests;
using OurPresence.Modeller.Interfaces;
using NSubstitute;
using System.Linq;
using OurPresence.Modeller.Generator;
using Xunit;

namespace OurPresence.Modeller.CoreFunctionalTests
{
    public class RcmsTests
    {
        [Fact]
        public void GivenDomainProjectGenerator_WhenGenerating_ThenProjectCreated()
        {
            var packageService = Substitute.For<IPackageService>();
            var settings = Substitute.For<ISettings>();
            settings.SupportRegen.Returns(true);
            var packages = new Packages(packageService, settings);
            settings.Packages.Returns(packages);
            var module = RcmsModuleBuilders.CreateProject();

            var c = new DomainProject.Generator(settings, module);

            IProject output = c.Create() as IProject;
            Approvals.VerifyAll(output.FileGroups.SelectMany(f=>f.Files), "file", f => $"{f.FullName}\r\n{f.Content}");
        }
    }
}
