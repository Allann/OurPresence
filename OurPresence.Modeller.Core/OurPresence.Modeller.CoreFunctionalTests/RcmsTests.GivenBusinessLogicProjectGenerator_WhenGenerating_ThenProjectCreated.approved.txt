file[0] = Nhvr.Rcms.BusinessLogic.csproj
<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <TargetFramework>net5.0</TargetFramework>
        <RootNamespace>Nhvr.Rcms.BusinessLogic</RootNamespace>
        <LangVersion>Preview</LangVersion>
        <Nullable>enable</Nullable>
    </PropertyGroup>

    <ItemGroup>
        <PackageReference Include="MediatR" Version="9.0.0" />
    </ItemGroup>

    <ItemGroup>
        <ProjectReference Include="..\Nhvr.Rcms.Common\Nhvr.Rcms.Common.csproj" />
    </ItemGroup>

</Project>

file[1] = usings.cs
global using System;
global using System.Collections.Generic;
global using System.Linq;
global using System.Text;
global using System.Threading.Tasks;
global using CSharpFunctionalExtensions;
global using MediatR;

global using Nhvr.Rcms.Common.Enums;

file[2] = ActivityLog\Officer\ActivityLog.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.ActivityLog.Officer
{
    public partial record ActivityLogOfficerRequest : IRequest<IEnumerable<ActivityLogOfficerResult>>
    { }

    public partial record ActivityLogOfficerResult
    {
        public int Id { get;  set; }

        public DateTimeOffset StartTime { get;  set; }

        public DateTimeOffset EndTime { get;  set; }

        public string Source { get;  set; }

        public string ActivityType { get;  set; }

        public string Remark { get;  set; }

        public bool Light { get;  set; }

        public bool Siren { get;  set; }

        public bool UrgentDutyDriving { get;  set; }

        public bool IsEditable { get;  set; }

        public bool SystemGenerated { get;  set; }

        public DateTimeOffset DateAdded { get;  set; }

    }
}

file[3] = ActivityLog\Add\ActivityLog.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.ActivityLog.Add
{
    public partial record ActivityLogAddRequest : IRequest<Result<ActivityLogAddResult>>
    {
        public DateTimeOffset DateAdded { get;  set; }

        public DateTimeOffset StartTime { get;  set; }

        public string ActivityType { get;  set; }

    }
    public partial record ActivityLogAddResult
    {
        public int Id { get;  set; }

    }
}

file[4] = ActivityLog\Update\ActivityLog.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.ActivityLog.Update
{
    public partial record ActivityLogUpdateRequest : IRequest<Result<ActivityLogUpdateResult>>
    {
        public DateTimeOffset EndTime { get;  set; }

        public string Remark { get;  set; }

        public bool? Light { get;  set; }

        public bool? Siren { get;  set; }

        public bool? UrgentDutyDriving { get;  set; }

    }
    public partial record ActivityLogUpdateResult
    {
        public int Id { get;  set; }

    }
}

file[5] = ActivityLog\Delete\ActivityLog.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.ActivityLog.Delete
{
    public partial record ActivityLogDeleteRequest : IRequest
    {
        public int Id { get;  set; }

    }
}

file[6] = Alert\Save\Alert.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.Alert.Save
{
    public partial record AlertSaveRequest : IRequest<Result<AlertSaveResult>>
    {
        public int Id { get;  set; }

        public int AlertLevel { get;  set; }

        public DateTimeOffset EffectiveDate { get;  set; }

        public DateTimeOffset ExpiryDate { get;  set; }

        public string Remark { get;  set; }

        public bool Silent { get;  set; }

        public int SilentNotifyContactId { get;  set; }

        public int PersonId { get;  set; }

        public int VehicleId { get;  set; }

        public int OrganisationId { get;  set; }

    }
    public partial record AlertSaveResult
    {
        public int Id { get;  set; }

    }
}

file[7] = Alert\Delete\Alert.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.Alert.Delete
{
    public partial record AlertDeleteRequest : IRequest
    {
        public int Id { get;  set; }

        public string Reason { get;  set; }

    }
}

file[8] = BatchProcess\SilentEmail\BatchProcess.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.BatchProcess.SilentEmail
{
    public partial record BatchProcessSilentEmailRequest : IRequest
    { }

}

file[9] = BatchProcess\InterceptEmail\BatchProcess.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.BatchProcess.InterceptEmail
{
    public partial record BatchProcessInterceptEmailRequest : IRequest
    { }

}

file[10] = BatchProcess\InterceptFile\BatchProcess.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.BatchProcess.InterceptFile
{
    public partial record BatchProcessInterceptFileRequest : IRequest
    { }

}

file[11] = BatchProcess\FinesVicRoad\BatchProcess.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.BatchProcess.FinesVicRoad
{
    public partial record BatchProcessFinesVicRoadRequest : IRequest
    { }

}

file[12] = BatchProcess\SqlMaintenance\BatchProcess.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.BatchProcess.SqlMaintenance
{
    public partial record BatchProcessSqlMaintenanceRequest : IRequest
    { }

}

file[13] = BatchProcess\TasInfringmentEmail\BatchProcess.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.BatchProcess.TasInfringmentEmail
{
    public partial record BatchProcessTasInfringmentEmailRequest : IRequest
    { }

}

file[14] = Organisation\Search\Organisation.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.Organisation.Search
{
    public partial record OrganisationSearchRequest : IRequest<Result<IEnumerable<OrganisationSearchResult>>>
    {
        public string Name { get;  set; }

        public string? ACN { get;  set; }

    }
    public partial record OrganisationSearchResult
    {
        public string Name { get;  set; }

        public string? ACN { get;  set; }

        public int? AlertLevel { get;  set; }

        public int? AssociatedAlertLevel { get;  set; }

        public string Source { get;  set; }

        public string Type { get;  set; }

        public string Status { get;  set; }

    }
}

file[15] = Organisation\Index\Organisation.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.Organisation.Index
{
    public partial record OrganisationByIdRequest : IRequest<Result<OrganisationDetailResult>>
    {
        public string Id { get;  set; }

        public string Juro { get;  set; }

        public bool? IncludeSilentAlert { get;  set; }

    }
    public partial record OrganisationDetailResult
    {
        public string Name { get;  set; }

        public string? ACN { get;  set; }

        public int? AlertLevel { get;  set; }

        public int? AssociatedAlertLevel { get;  set; }

        public string Source { get;  set; }

        public string Type { get;  set; }

        public string Status { get;  set; }

    }
}

file[16] = Vehicle\Create\Vehicle.generated.cs
// Auto-generated using OurPresence.Modeller template 'Business Logic Behaviour' version 1.0.0

// --------------------------------------------------------------------------------
// WARNING: This file will be overwritten if re-generation is triggered.
// --------------------------------------------------------------------------------

namespace Nhvr.Rcms.BusinessLogic.Vehicle.Create
{
    public partial record VehicleCreateRequest : IRequest
    {
        public States RegistrationState { get;  set; }

        public string? RegistrationNumber { get;  set; }

        public bool IsNationalRegistration { get;  set; }

        public string? VehicleIdentificationNumber { get;  set; }

        public string? EngineNumber { get;  set; }

        public string? ChassisNumber { get;  set; }

        public string SourceVehicleId { get;  set; }

        public SourceSystemTypes SystemId { get;  set; }

    }
}

